connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1592,6 +1704,16 @@ public void onSessionKeyUpdate(SessionKey sessionKey) {
                 }
             }
         }
+
+        @Override
+        public void onRestartRequest(RestartRequest request) {
+            log.info("Received {}", request);
```

Comment:
I assume the choice of INFO here was intentional. If not, at what log level do we really want this log message? If so or if we do want this at INFO level, then maybe we should modify the `RestartRequest.toString()` method to be a little more user friendly (e.g., `restart request for connector %s (...)`).

And would it help to mention here that it's being enqueued? Maybe:
```suggestion
            log.info("Received and enqueuing {}", request);
```